File: programmers/04.두 정수 사이의 합.js
// a와 b가 같은 경우는 둘 중 아무 수나 리턴하세요.
// a와 b는 -10,000,000 이상 10,000,000 이하인 정수입니다.
// a와 b의 대소관계는 정해져있지 않습니다.

// a와 b사이에 속한 모든 정수의 합
function solution(a,b) {
    let answer = 0;

    if(a<=b) { // 같을때도
        for(let i=a; i<=b; i++) {
            answer = answer + i;
        }
    }
    if(a>b) {
        for(let i=b; i<=a; i++) {
            answer = answer + i;
        }
    }

    return answer;

    // return (a+b)*(Math.abs(b-a)+1)/2; 다른 사람 코드
    // abs 절대값 반환
}

console.log(solution(10,5));

File: programmers/swift/04. 두 정수 사이의 합.swift
func solution(_ a:Int, _ b:Int) -> Int64 {  //// Int64
    var answer : Int = 0
    
    for i in a < b ? a...b : b...a {
        answer += i
    }
        
    return Int64(answer)
}


// let numbers = [1, 2, 3, 4]
// let numberSum = numbers.reduce(0, { x, y in
//     x + y
// })
// numberSum == 10

/*
* reduce: 내부의 값들을 결합
*/

// for-in
let numberArray = [1,2,3,4,5]
var sum = 0
for number in numberArray{
  sum += number
}
print(sum) // 15

// reduce
let numberArray = [1,2,3,4,5]
let sum = numberArray.reduce(0){ $0 + $1 }
let subract = numberArray.reduce(0){ $0 - $1 }
print(sum) // 15
print(subract) // -15
